src/main/java/net/glowstone/command/minecraft/CloneCommand.java

```diff
@@ -0,0 +1,338 @@
+
+import com.google.common.collect.AbstractIterator;
+import net.glowstone.GlowWorld;
+import net.glowstone.block.GlowBlock;
+import net.glowstone.block.entity.BlockEntity;
+import net.glowstone.command.CommandUtils;
+import net.glowstone.constants.ItemIds;
+import net.glowstone.util.nbt.CompoundTag;
+import org.bukkit.ChatColor;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.command.CommandSender;
+import org.bukkit.command.defaults.VanillaCommand;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.Iterator;
+
+public class CloneCommand extends VanillaCommand {
+    public enum MaskMode {
+        REPLACE("replace"),
+        MASKED("masked"),
+        FILTERED("filter");
+
+        private final String commandName;
+
+        MaskMode(String commandName) {
+            this.commandName = commandName;
+        }
+
+        public String getCommandName() {
+            return commandName;
+        }
+
+        public static MaskMode fromCommandName(String commandName) {
+            for (MaskMode maskMode : values()) {
+                if (maskMode.getCommandName().equals(commandName)) {
+                    return maskMode;
+                }
+            }
+            return null;
+        }
+    }
+
+    public enum CloneMode {
+        NORMAL("normal", false),
+        FORCE("force", true),
+        MOVE("move", false);
+
+        private final String commandName;
+        private final boolean allowedToOverlap;
+
+        CloneMode(String commandName, boolean allowedToOverlap) {
+            this.commandName = commandName;
+            this.allowedToOverlap = allowedToOverlap;
+        }
+
+        public String getCommandName() {
+            return commandName;
+        }
+
+        public boolean isAllowedToOverlap() {
+            return allowedToOverlap;
+        }
+
+        public static CloneMode fromCommandName(String commandName) {
+            for (CloneMode cloneMode : values()) {
+                if (cloneMode.getCommandName().equals(commandName)) {
+                    return cloneMode;
+                }
+            }
+            return null;
+        }
+    }
+
+    public CloneCommand() {
+        super(
+                "clone",
+                "Clones a section of the world.", "/clone <x1> <y1> <z1>  <x2> <y2> <z2>  <x> <y> <z> [maskMode] [cloneMode] [tileName] [tileData]",
+                Collections.emptyList()
+        );
+        setPermission("minecraft.command.seed");
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String label, String[] args) {
+        if (!testPermission(sender)) {
+            return false;
+        }
+
+        if (args.length < 9) {
+            sender.sendMessage(ChatColor.RED + "Usage: " + usageMessage);
+            return false;
+        }
+
+        if (!CommandUtils.isPhysical(sender)) {
+            sender.sendMessage("Cannot access blocks outside of the world");
```

Comment:
I think the error message and the if-statement don't match here. (e.g. should be `This command may only be executed by physical objects` or something. I think other commands have this message somewhere.